common/src/main/java/io/homeassistant/companion/android/common/sensors/NetworkSensorManager.kt

```diff
@@ -570,6 +600,79 @@ class NetworkSensorManager : SensorManager {
         })
     }
 
+    @SuppressLint("MissingPermission")
+    private fun updateCellConnectionSensor(context: Context) {
+        if (!isEnabled(context, cellConnection) || !hasTelephony(context)) {
+            return
+        }
+
+        val telephonyManager = context.getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
+
+        val cell = telephonyManager.getAllCellInfo().findLast { cell -> cell.isRegistered } ?: return
```

Comment:
Does this line mean not currently connected to any cell tower? If so `STATE_UNAVAILABLE` may make more sense here because we should report if the user has telephony but may be out of cell service or has airplane mode with wifi connected.